### DISCUSSION TITLE
Adding new libraries that does not starts with "lib"

### DISCUSSION THREAD

**Discussion by xshuqun** [eclipse-cdt/cdt#639] [Q&A] [answered]
Hi All,

Basically I have two libraries, lets name it one.a and two.a
In GCC, I had used the following way to add into makefiles for build, and it works well  with no build error.
`-Wl,--start-group <path-to>/one.a -Wl,--end-group`
`-Wl,--start-group <path-to>/two.a -Wl,--end-group`

But I tried to add it into the eclipse project using my Ubuntu machine, there will be build errors like "undefined reference", which are all within the two libraries. 
I added the libraries under Project Properties > C/C++ Build > Settings > Cross GCC Linker > Miscellaneous > Linker flags.

Can someone give an advise on it?

Thank you.

**Comment by jld01** [comment, marked as answer]
Try adding your library file names to the _Libraries_ list using a `:` prefix and add the folders containing your library files to the _Library search path_ list as follows:

![image](https://github.com/eclipse-cdt/cdt/assets/45816183/b00c00b7-490f-4062-8298-93a0b6f65388)

**Reply by xshuqun** [reply]
Thank you @jld01 that solves my problem!
Do you have any idea how output files are being handled, where are the settings?
In my project, it outputs .d, .o, .i and .s files. The .d and .o are being placed in its repective folder which the original .c files are located, but .i and .s are placed outside, which is very messy, I need to organize it and place it in the respective folders declared.

**Reply by jld01** [reply]
As a general rule, please create a _new_ GitHub discussion for each topic.

I think you need to use the GCC flag `-save-temps=obj` instead of `-save-temps`.
